modules/planning/open_space/trajectory_smoother/dual_variable_warm_start_problem.cc

```diff
@@ -41,94 +41,117 @@ bool DualVariableWarmStartProblem::Solve(
-      new DualVariableWarmStartIPOPTInterface(
-          horizon, ts, ego, obstacles_edges_num, obstacles_num, obstacles_A,
-          obstacles_b, xWS, planner_open_space_config_);
 
-  DualVariableWarmStartIPOPTQPInterface* ptop2 =
-      new DualVariableWarmStartIPOPTQPInterface(
-          horizon, ts, ego, obstacles_edges_num, obstacles_num, obstacles_A,
-          obstacles_b, xWS, planner_open_space_config_);
-
-  Ipopt::SmartPtr<Ipopt::TNLP> problem = ptop;
-  if (planner_open_space_config_.dual_variable_warm_start_config().
-      qp_format()) {
-    problem = ptop2;
-  }
-  // Create an instance of the IpoptApplication
-  Ipopt::SmartPtr<Ipopt::IpoptApplication> app = IpoptApplicationFactory();
-
-  app->Options()->SetIntegerValue("print_level", planner_open_space_config_.
-      dual_variable_warm_start_config().ipopt_config().ipopt_print_level());
-  app->Options()->SetIntegerValue("mumps_mem_percent",
-      planner_open_space_config_.dual_variable_warm_start_config().
-      ipopt_config().mumps_mem_percent());
-  app->Options()->SetNumericValue("mumps_pivtol", planner_open_space_config_.
-      dual_variable_warm_start_config().ipopt_config().mumps_pivtol());
-  app->Options()->SetIntegerValue("max_iter", planner_open_space_config_.
-      dual_variable_warm_start_config().ipopt_config().ipopt_max_iter());
-  app->Options()->SetNumericValue("tol", planner_open_space_config_.
-      dual_variable_warm_start_config().ipopt_config().ipopt_tol());
-  app->Options()->SetNumericValue("acceptable_constr_viol_tol",
-      planner_open_space_config_.dual_variable_warm_start_config().
-      ipopt_config().ipopt_acceptable_constr_viol_tol());
-  app->Options()->SetNumericValue("min_hessian_perturbation",
-      planner_open_space_config_.dual_variable_warm_start_config().
-      ipopt_config().ipopt_min_hessian_perturbation());
-  app->Options()->SetNumericValue("jacobian_regularization_value",
-      planner_open_space_config_.dual_variable_warm_start_config().
-      ipopt_config().ipopt_jacobian_regularization_value());
-  app->Options()->SetStringValue("print_timing_statistics",
-      planner_open_space_config_.dual_variable_warm_start_config().
-      ipopt_config().ipopt_print_timing_statistics());
-  app->Options()->SetStringValue("alpha_for_y",
-      planner_open_space_config_.dual_variable_warm_start_config().
-      ipopt_config().ipopt_alpha_for_y());
-  app->Options()->SetStringValue("recalc_y", planner_open_space_config_.
-      dual_variable_warm_start_config().ipopt_config().ipopt_recalc_y());
-  // for qp problem speed up
   if (planner_open_space_config_.dual_variable_warm_start_config().
-      qp_format()) {
-    app->Options()->SetStringValue("mehrotra_algorithm", "yes");
-  }
-
-  Ipopt::ApplicationReturnStatus status = app->Initialize();
-  if (status != Ipopt::Solve_Succeeded) {
-    AERROR
-        << "*** Dual variable wart start problem error during initialization!";
+      enable_osqp()) {
```

Comment:
use an enum to chose between "OSQP", "IPOPT", and "IPOPTQP" please.

Reply:
> use an enum to chose between "OSQP", "IPOPT", and "IPOPTQP" please.

It is done.

Reply:
use ENUM, not number